### ISSUE TITLE
Implement move_t struct and functions from last week's design document.

### ISSUE THREAD

**Issue by nicholaswlee** [uchicago-cs/chiventure#1354]
Here is the [doc](https://github.com/uchicago-cs/chiventure/wiki/Battles-~-Move-Changes-Design-Document).
This is what the move_t struct looks like:
`typedef struct move_t {`
     
     int id; //the unique identifier of the move

     char* name; //the name of the move

     char* info; //a description of what the move does

     damage_type_t dmg_type; //whether a move does physical damage, magical damage, or no damage

     target_type_t stat_mods; //whether a move changes a user's stats, the opponent's stats, both, or neither

     target_type_t effects; //whether a move has an effect that targets the user, opponent, both, or neither

     target_count_t count; 

     int sp_cost; //the amount of sp required to use a move

     battle_item_t* req_item; //an item required to use the move or NULL for none

     int damage; //the damage the move will do

     int accuracy; //the accuracy of the move

     stat_change_t* user_mods; //stat changes for the user

     stat_change_t* opponent_mods; //stat changes for the opponent

     move_t* prev; //the previous move in the list, or NULL for no move

     move_t* next; //the next move in the list, or NULL for no move
`} move_t;`

Our goal is to change the battle module to accommodate this struct. This will allow moves to be extremely customizable allowing them to do stat changes for both the user and the opponent and do damaging moves that can also change stats. Our branch's name is: battle-move-changes

**Comment by corian-jal** [comment]
**Division of Work**
_battle_flow.c -- Nicholas_
battle_flow_move()
enemy_make_move()
calculate_accuracy()

_battle_print.c -- Nicholas
print_battle_move()
print_battle_miss()

_battle_ai.c -- Fatimah_
damage()

_battle_default_object.c -- Fatimah_
get_random_default_move()

_battle_moves.c -- Nafis_
move_init()
move_new()
move_free()
(example moves)

**Comment by nicholaswlee** [comment]
We have done all of the implementation, however there are a couple or errors in compiling. This will be fixed throughout the next few days.

**Comment by nicholaswlee** [comment]
We have been fixing compilation errors throughout this evening. We have split up the work depending on which files have errors.

**Comment by nicholaswlee** [comment]
We came across an issue with player Classes's move_list. To fix this problem, in our translation we have set default values such as PHYS, 100 accuracy, etc.

**Comment by nicholaswlee** [comment]
We have been fixing our tests to accommodate for move_t changes. We will need to create another PR that will handle further testing for things like stat changes and physical/magical moves.

**Comment by nicholaswlee** [comment]
We had to change some tests in the following files that were not under battle module:
In NPC:
`test_npc.c`
`test_npc_battle.c`
In Skill Trees:
`effect_test.c`

**Comment by nicholaswlee** [comment]
We have completed our implementation and have submitted a pull request. We have made a couple new functions as outlined in our doc and above. The biggest change is that there are different types of moves that allow for more customizable battles. All of our tests have passed, and we have implemented our code as laid out in the [design document](https://github.com/uchicago-cs/chiventure/wiki/Battles-~-Move-Changes-Design-Document). Here is the [PR](https://github.com/uchicago-cs/chiventure/pull/1381).
